DataManagementSystem/DB/FileCatalogComponents/DirectoryTreeBase.py

```diff
@@ -544,184 +547,189 @@ def getDirectoryPermissions( self, path, credDict ):
     if self.db.globalReadAccess:
       resultDict['Read'] = True
     else:
-      resultDict['Read'] = ( owner and mode & stat.S_IRUSR > 0 )\
-                           or ( group and mode & stat.S_IRGRP > 0 )\
-                           or mode & stat.S_IROTH > 0
-                           
-    resultDict['Write'] = ( owner and mode & stat.S_IWUSR > 0 )\
-                          or ( group and mode & stat.S_IWGRP > 0 )\
-                          or mode & stat.S_IWOTH > 0
+      resultDict['Read'] = (owner and mode & stat.S_IRUSR > 0)\
+          or (group and mode & stat.S_IRGRP > 0)\
+          or mode & stat.S_IROTH > 0
+
+    resultDict['Write'] = (owner and mode & stat.S_IWUSR > 0)\
+        or (group and mode & stat.S_IWGRP > 0)\
+        or mode & stat.S_IWOTH > 0
 
-    resultDict['Execute'] = ( owner and mode & stat.S_IXUSR > 0 )\
-                            or ( group and mode & stat.S_IXGRP > 0 )\
-                            or mode & stat.S_IXOTH > 0
+    resultDict['Execute'] = (owner and mode & stat.S_IXUSR > 0)\
+        or (group and mode & stat.S_IXGRP > 0)\
+        or mode & stat.S_IXOTH > 0
 
-    return S_OK( resultDict )
+    return S_OK(resultDict)
 
-  def getFileIDsInDirectoryWithLimits( self, dirID, credDict, startItem = 1, maxItems = 25 ):
+  def getFileIDsInDirectoryWithLimits(self, dirID, credDict, startItem=1, maxItems=25):
     """ Get file IDs for the given directory
     """
     dirs = dirID
-    if type( dirID ) != ListType:
+    if not isinstance(dirID, ListType):
```

Comment:
You can move all these to `isinstance(dirID, list)` and then simply remove the import of `types`